cfn-resources/serverless-instance/cmd/resource/resource.go

```diff
@@ -104,20 +107,16 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	setup()
 
-	_, _ = log.Debugf("Read() currentModel:%+v", currentModel)
-
 	// Validation
 	modelValidation := validateModel(ReadRequiredFields, currentModel)
 	if modelValidation != nil {
 		return *modelValidation, nil
 	}
 
 	// Create atlas client
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		_, _ = log.Warnf("Read - error: %+v", err)
-		return progress_events.GetFailedEventByCode(fmt.Sprintf("Error creating mongoDB client : %s", err.Error()),
-			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	client, peErr := util.NewMongoDBClient(req, currentModel.Profile)
```

Comment:
Missing profile read?

```suggestion
        if currentModel.Profile == nil {
		currentModel.Profile = pointy.String(util.DefaultProfile)
	}
	
	client, peErr := util.NewMongoDBClient(req, currentModel.Profile)
```

Reply:
Also, wondering should this instead happen in the `util. NewMongoDBClient` as the fallback behaviour is needed everytime and is repeating a lot.

Reply:
i was thinking about that, but all the currentmodels are different, we might need to do some overengineering to do this on the NewMongoClient

Reply:
Ack, we can reconsider in the future when all the CFN changes are in place and review usage in a single block.